internal/datastore/common/sql.go

```diff
@@ -124,24 +112,22 @@ func (sqf SchemaQueryFilterer) FilterToSubjectFilter(filter *v1.SubjectFilter) S
 		sqf.tracerAttributes = append(sqf.tracerAttributes, SubObjectIDKey.String(filter.OptionalSubjectId))
 	}
 
-	sqf.currentEstimatedSize += len(filter.SubjectType) + len(filter.OptionalSubjectId)
-
 	if filter.OptionalRelation != nil {
 		dsRelationName := stringz.DefaultEmpty(filter.OptionalRelation.Relation, datastore.Ellipsis)
 
 		sqf.queryBuilder = sqf.queryBuilder.Where(sq.Eq{sqf.schema.ColUsersetRelation: dsRelationName})
 		sqf.tracerAttributes = append(sqf.tracerAttributes, SubRelationNameKey.String(dsRelationName))
-		sqf.currentEstimatedSize += len(dsRelationName)
 	}
 
 	return sqf
 }
 
 // FilterToUsersets returns a new SchemaQueryFilterer that is limited to resources with subjects
-// in the specified list of usersets.
-func (sqf SchemaQueryFilterer) FilterToUsersets(usersets []*v0.ObjectAndRelation) SchemaQueryFilterer {
+// in the specified list of usersets. Nil or empty usersets parameter does not affect the underlying
+// query.
+func (sqf SchemaQueryFilterer) filterToUsersets(usersets []*v0.ObjectAndRelation) SchemaQueryFilterer {
 	if len(usersets) == 0 {
-		panic("Got empty usersets filter")
+		return sqf
```

Comment:
Why this change?

Reply:
not sure - looking at where it's used, we shouldn't ever be sending a batch of size 0. I'll play with it a bit.